src/Surfnet/StepupRa/RaBundle/Controller/RaaController.php

```diff
@@ -37,17 +38,17 @@ public function institutionConfigurationAction(Request $request)
         $identity = $this->getUser();
 
         $profile = $this->getProfileService()->findByIdentityId($identity->id);
-        $choices = $profile->getRaaInstitutions();
-        $institution = reset($choices);
+
+        if ($this->isGranted('ROLE_SRAA')) {
+            $institution = $identity->institution;
+            $choices = $this->getInstitutionListingService()->getAll();
```

Comment:
`getAll` can return `false` the code does not anticipates that possibility. Well it kind of does, as `count(false)` equals `int(1)`. Not something I'd like to rely on..

Reply:
Your own institution would always be listed but I removed the array_combine to always return an array to prevent bugs in the future.